spec/index.html

```diff
@@ -732,12 +741,15 @@ <h2>Literals</h2>
         not defined by this specification.</li>
     </ol>
 
-    <p><dfn data-local-lt="term-equal">Literal term equality</dfn>: Two literals are term-equal (the same
-      RDF literal) if and only if the two <a>lexical forms</a>,
-      the two <a>datatype IRIs</a>, and the two
-      <a>language tags</a> (if any) compare equal,
-      character by character. Thus, two literals can have the same value
-      without being the same RDF term. For example:</p>
+    <p><dfn data-local-lt="term-equal">Literal term equality</dfn>:
+      Two literals are term-equal (the same <a>RDF literal</a>)
+      if and only if the two <a>lexical forms</a>,
+      the two <a>datatype IRIs</a>,
+      the two <a>language tags</a> (if any), and
+      the two <a>text directions</a> (if any)
+      compare equal, character by character.
```

Comment:
We previously say that the value space of `language tag` is lower case, is it redundant to say to use a case insensitive comparison here?

Reply:
> is it redundant to say to use a case insensitive comparison here

I think it conforms to Postel's Law, will clearly reflect user intent, and will be better than case sensitivity-based errors.

Reply:
I was jumping to conclusions. Some investigation: it isn't as simple as "case insensitive".

The text above this paragraph says "MAY lower case" the concrete string that is the language tag.

The text here is about term-equality and does not say it is a value-space comparison. (FWIW "value space" for language tags is a bit meaningless - "value spaces" involve datatypes but we are where we are.)

The RDF 1.1 text:

> the two language tags (if any) compare equal, character by character

does allow `"abc"@en` and `"abc"@EN` as different terms, whether that was intended or not. 

The root problem is that RDF has not used the canonical form for language tags. Some users do care about this.

At users' request Jena has options to leave as-is, always lower-case and always canonicalization.

Maybe better to leave as "character by character" because otherwise it is a implementation change.

Reply:
Hm, yes, I see what you mean. Accidental inconsistency is a rarely vanquished hobgoblin, especially across specs developed separately over years. We have our work cut out for us, in trying to bring consistency to all these docs that we're simultaneously trying to upgrade/update.

"Character by character" is at least clear, and we can include a note that advises deployers of the potential need to enact Jena's options — i.e., keep original langtag casing, make all langtags lower-case, or (whatever you meant by "always canonicalization").

Reply:
We need something to reconcile the notions of a lower-case value (space) for the language-tag and the fact that it's compared character by character (code point by code point?). Are there systems where `"abc"@en` and `"abc"@EN` are not considered the same term.

It might say something like "two language tags (if any) compare equal after normalizing to lower case".

The sentence "The value space of language tags is always in lower case" might be changed to "The value of language tags is always treated as being in lower case".

Reply:
We should avoid "value" because that is about datatype literals.

"compare equal as if normalized to lower case".

which does not imply they are converted (the earlier text is MAY).

> Are there systems where "abc"@en and "abc"@EN
Jena keeps them apart but they are the same term (yes - that's contradictory)

It's about meeting the user expectation of round-trip with no change.